### ISSUE TITLE
pgagroal-cli reload and added server

### ISSUE THREAD

**Issue by fluca1978** [agroal/pgagroal#223]
Running `pgagroal` version 1.5.0.
TL;DR: if I add a new not-primary server to the configuration, issuing a `reload` makes the server appearing in the list of available (and I can `switch-to`). The old primary is then reported as `Not init` which is not the correct status.

Initial situation, one server with prefill:
```
[pgagroal]
host = *
port = 5432
unix_socket_dir = /tmp

log_type  = console
log_level = info
max_connections = 900
pipeline = performance

[venkman]
host = venkman
port = 5432
primary = on
```

and the prefill:

```
% cat /etc/pgagroal/pgagroal_databases.conf
# DATABASE USER MAX INITIAL MIN
 pgbench pgbench 600 200  5
```

When I start `pgagroal` the prefill works, and the connections are established:

```
% pgagroal-cli details | less
Status:              Running
Active connections:  0
Total connections:   200
Max connections:     900
---------------------
Server:              venkman
Host:                venkman
Port:                5432
State:               Primary
---------------------
Database:            pgbench
Username:            pgbench
Active connections:  0
Max connections:     600
Initial connections: 200
Min connections:     5
```

So far so good.
Now adding a new server in the configuration:

```
% cat /etc/pgagroal/pgagroal.conf
...
[venkman]
host = venkman
port = 5432
primary = on

[spengler]
host = spengler
port = 5432
primary = off
```

if I now issue a `reload` the `details` command reports that even the primary is in state not initialized:

```
% pgagroal-cli reload
% pgagroal-cli details | less
Status:              Running
Active connections:  0
Total connections:   200
Max connections:     900
---------------------
Server:              venkman
Host:                venkman
Port:                5432
State:               Not init
---------------------
Server:              spengler
Host:                spengler
Port:                5432
State:               Not init
---------------------
Database:            pgbench
Username:            pgbench
Active connections:  0
Max connections:     600
Initial connections: 200
Min connections:     5
```

The server `venkman`, that holds the connections, is now `Not init` and this is not correct.
In the logs there is a line:

```
INFO  configuration.c:2738 Restart required for pidfile - Existing /tmp/pgagraol.5432.pid New
```

This is strange: the new server has been added to the list, so `reload` partially worked but the old primary is not in the correct status, so I guess if this is not a reloadable option the old configuration should be retained.
Am I missing something?

**Comment by jesperpedersen** [comment]
Yeah, it sounds like there is a couple of bugs in that work-flow. Thanks for the report !

**Comment by fluca1978** [comment]
Seems the problem is in `copy_server`, that according to me is used only when moving configuration from one part to another due to a `reload`: <https://github.com/agroal/pgagroal/blob/master/src/libpgagroal/configuration.c#L2701>. However, the `reload` configuration has already the `state` set to `SERVER_NOTINIT_PRIMARY` instead of `SERVER_PRIMARY`, as it was.

**Comment by jesperpedersen** [comment]
Yeah, existing state needs to be considered

**Comment by fluca1978** [comment]
I don't understand why, when `copy_server` is invoked, the original configuration has already the primary server marked as `SERVER_NOTINIT_PRIMARY`, so the commit makes the new configuration primary server to such state. This is better than `SERVER_NOTINIT`, but I think the original configuration is "reset" too early. I need a little help here to understand if this is correct and where it happens.

**Comment by fluca1978** [comment]
Close by work in #233
